Add quiet mode

jspeed-meyers opened the pull request:
Fix #57

Add an output option of `quiet` optimized for using this tool in CI. Return 0 if successful and 1 otherwise.

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

Commit:
Add quiet mode

Add an output option of bool to enable a "quiet" mode where
the tool prints 0 if the SBOM is ntia conformant and -1
otherwise.

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

anthonyharrison reviewed (changes requested):
Suggest change option from 'bool' to 'quiet'

Change 

```
if sbom.check_ntia_minimum_elements_compliance():
     print(0)
else:
    print(-1)
```
To

 `sys.exit(0 if sbom.check_ntia_minimum_elements_compliance()  else -1)`

zvr commented:
Please use exit code `0` on success and `1` on error, *not* `-1` which is system-dependent and many systems only support unsigned values.

Ideally, you could also use other error `sysexits.h` for more specific errors.

jspeed-meyers commented:
> Please use exit code 0 on success and 1 on error, not -1 which is system-dependent and many systems only support unsigned values.

Thank you, @zvr. I'll change the code to do that.

@anthonyharrison -- thanks for the better syntax and `quiet` naming suggestion. I'll change that. New commit coming!

jspeed-meyers commented:
> Ideally, you could also use other error sysexits.h for more specific errors.

I was unfamiliar with `sysexits.h`. Thank you, @zvr. I am currently thinking to stick with `sys.exit(0)` and `sys.exit(1)` since, I think, it will be more familiar to Python programmers. Please object if you think this decision is off-base.

Commit:
Use sys.exit error codes

Add sys.exit error codes instead of print statements and also use
"quiet" as option name

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

Commit:
Use oneliner for if statement

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

Commit:
Use 1, not -1 for error exit

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

Commit:
Add quiet explanation to README

Add quiet usage explanation to README and fix typos earlier in README

Signed-off-by: John Speed Meyers <jsmeyers@chainguard.dev>

zvr commented:
Sure, no issue only having success (`0`) and a single error (`1`) exit codes.

anthonyharrison reviewed (changes requested):
I think the sys.exit should always be called and not just in the 'quiet' operation. 

Can then get rid of the errorneous line 49  which should have been  `if output == "quiet"`

Commit:
Make all branches execute sys.exit

jspeed-meyers commented:
Good call, @anthonyharrison

jspeed-meyers commented:
Thank you again, @linynjosh!

Commit:
Merge branch 'main' into quiet_option
